backend/user/notifications.py

```diff
@@ -0,0 +1,56 @@
+import collections
+import os
+
+from apns2.client import APNsClient
+from apns2.credentials import TokenCredentials
+from apns2.payload import Payload
+
+
+# taken from the apns2 method for batch notifications
+Notification = collections.namedtuple("Notification", ["token", "payload"])
+
+
+def send_push_notif(token, title, body, isDev=False):
+    """Sends single push notification to token object"""
+
+    client = get_client(isDev)
+    alert = {"title": title, "body": body}
+    payload = Payload(alert=alert, sound="default", badge=0)
+    topic = "org.pennlabs.PennMobile" + (".dev" if isDev else "")
+    client.send_notification(token.token, payload, topic)
+
+
+def send_push_notif_batch(tokens, title, body, isDev=False):
+    """Sends push notifications to group of token objects"""
+
+    client = get_client(isDev)
+    alert = {"title": title, "body": body}
+    payload = Payload(alert=alert, sound="default", badge=0)
+    notifications = [Notification(token.token, payload) for token in tokens]
+    topic = "org.pennlabs.PennMobile" + (".dev" if isDev else "")
+    client.send_notification_batch(notifications=notifications, topic=topic)
+
+
+def send_shadow_push_notif_batch(tokens, body, isDev=False):
+    """Sends shadow push notifications to group of token objects"""
+
+    client = get_client(isDev)
+    payload = Payload(content_available=True, custom=body)
+    notifications = [Notification(token.token, payload) for token in tokens]
+    topic = "org.pennlabs.PennMobile" + (".dev" if isDev else "")
+    client.send_notification_batch(notifications=notifications, topic=topic)
+
+
+def get_client(isDev):
+    """Creates and returns APNsClient based on iOS credentials"""
+
+    auth_key_path = os.path.join(
```

Comment:
Can you make a Shortcut story on this line? Don't know if this will be a problem, but not sure if the file path is the same in production as it was in labs-api-server. If notifications works fine, that's great! Otherwise, this might be the root of the bug